src/Parser/Hunter/Survival/Modules/Spells/SerpentSting.js

```diff
@@ -3,33 +3,193 @@ import React from 'react';
+  accumulatedPercentRemainingOnRefresh = 0;
+  hasVV = false;
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.SERPENT_STING_SV.id) {
+      return;
+    }
+    this.casts++;
+    if (this.selectedCombatant.hasBuff(SPELLS.VIPERS_VENOM_BUFF.id)) {
+      this.hasVV = true;
+    }
+  }
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.SERPENT_STING_SV.id) {
+      return;
+    }
+    this.bonusDamage += event.amount + (event.absorbed || 0);
+  }
+
+  on_byPlayer_applydebuff(event) {
+    const spellId = event.ability.guid;
+    let targetInstance = event.targetInstance;
+    if (spellId !== SPELLS.SERPENT_STING_SV.id) {
+      return;
+    }
+    if (targetInstance === undefined) {
+      targetInstance = 1;
+    }
+    const hastedSerpentStingDuration = BASELINE_DURATION / (1 + this.statTracker.currentHastePercentage);
+    const serpentStingTarget = { targetID: event.targetID, targetInstance: targetInstance, timestamp: event.timestamp, serpentStingDuration: hastedSerpentStingDuration };
+    this.serpentStingTargets.push(serpentStingTarget);
+
+    this.hasVV = false;
+  }
+
+  on_byPlayer_removedebuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.SERPENT_STING_SV.id) {
+      return;
+    }
+    for (let i = 0; i < this.serpentStingTargets.length; i++) {
+      if (event.timestamp - this.serpentStingTargets[i].timestamp > this.serpentStingTargets[i].serpentStingDuration) {
+        this.serpentStingTargets.splice(i, 1);
+      }
+    }
+  }
+
+  on_byPlayer_refreshdebuff(event) {
+    const spellId = event.ability.guid;
+    let targetInstance = event.targetInstance;
+    if (spellId !== SPELLS.SERPENT_STING_SV.id) {
+      return;
+    }
+    for (let i = 0; i < this.serpentStingTargets.length; i++) {
+      if (event.timestamp - this.serpentStingTargets[i].timestamp > this.serpentStingTargets[i].serpentStingDuration) {
+        this.serpentStingTargets.splice(i, 1);
+      }
+    }
+    if (this.serpentStingTargets.length === 0) {
+      return;
+    }
+    this.timesRefreshed++;
+    if (targetInstance === undefined) {
+      targetInstance = 1;
+    }
+    const hastedSerpentStingDuration = BASELINE_DURATION / (1 + this.statTracker.currentHastePercentage);
+    const serpentStingTarget = { targetID: event.targetID, targetInstance: targetInstance, timestamp: event.timestamp };
+    for (let i = 0; i <= this.serpentStingTargets.length - 1; i++) {
+      if (this.serpentStingTargets[i].targetID === serpentStingTarget.targetID && this.serpentStingTargets[i].targetInstance === serpentStingTarget.targetInstance) {
+        const timeRemaining = this.serpentStingTargets[i].serpentStingDuration - (event.timestamp - this.serpentStingTargets[i].timestamp);
+        if (timeRemaining < (this.serpentStingTargets[i].serpentStingDuration * PANDEMIC) && !this.hasVV) {
+          this.badRefresh++;
```

Comment:
Shouldn't this be `timeRemaining > (hastedSerpentStingDuration * PANDEMIC)`?

Reply:
Good catch! It should indeed. Fixed.

Reply:
Using `hastedSerpentStingDuration` is also important imo, as that's what is used by the refreshing logic (just below).

Reply:
Like the latest commit added right?